streams/src/test/java/org/apache/kafka/streams/kstream/RepartitionTopicNamingTest.java

```diff
@@ -496,7 +496,8 @@ private Topology buildTopology(final String optimizationConfig) {
         mappedStream.filter((k, v) -> k.equals("A"))
                 .join(countStream, (v1, v2) -> v1 + ":" + v2.toString(),
                         JoinWindows.of(Duration.ofMillis(5000L)),
-                        Joined.with(Serdes.String(), Serdes.String(), Serdes.Long(), fourthRepartitionTopicName))
+                        Joined.with(Serdes.String(), Serdes.String(), Serdes.Long(), fourthRepartitionTopicName),
+                    Materialized.as(fourthRepartitionTopicName))
```

Comment:
Why do we need to name the store in this test? If the test fails without, it seem to indicate a backward incompatible change?

Reply:
I don't think it's a backward-incompatible change.  The name of the store in the topology changed as we no longer use `Joined` to name the store, so the name change is expected.  Using `Joined` to name the store is only on trunk ATM, and merging KIP-479 will keep users upgrading from 2.3 -> 2.4 with named repartition topics from getting a changed store name without explicitly setting it.

Reply:
:+1: